app/controllers/cookies_controller.rb

```diff
@@ -0,0 +1,55 @@
+class CookiesController < BasePublicController
+  include JourneyConcern
+
+  def accept
+    cookies.encrypted[:accept_cookies] = {
+      value: {state: true, message: true}.to_json,
+      expires: 90.days.from_now
+    }
+
+    respond_to do |format|
+      format.js
+      format.html { redirect_to request.env["HTTP_REFERER"] }
+    end
+  end
+
+  def reject
+    cookies.encrypted[:accept_cookies] = {
+      value: {state: false, message: true}.to_json,
+      expires: 90.days.from_now
+    }
+
+    respond_to do |format|
+      format.js
+      format.html { redirect_to request.env["HTTP_REFERER"] }
+    end
+  end
+
+  def hide
+    state = JSON.parse(cookies.encrypted[:accept_cookies])["state"]
+
+    cookies.encrypted[:accept_cookies] = {
+      value: {state:, message: false}.to_json,
+      expires: 90.days.from_now
+    }
+
+    redirect_to request.env["HTTP_REFERER"]
+  end
```

Comment:
(not a blocker) could these actions just use `update` passing different params?

Reply:
could be, i think it would shift more code to the front. i don't think it matters hugely and expect this code to be pretty stagnant